addons/account/models/account_account.py

```diff
@@ -430,24 +457,36 @@ def code_is_available(new_code):
 
         raise UserError(_('Cannot generate an unused account code.'))
 
+    @api.depends_context('allowed_company_ids')
     def _compute_current_balance(self):
         balances = {
             account.id: balance
             for account, balance in self.env['account.move.line']._read_group(
-                domain=[('account_id', 'in', self.ids), ('parent_state', '=', 'posted')],
+                domain=[
+                    ('account_id', 'in', self.ids),
+                    ('parent_state', '=', 'posted'),
+                    *self.env['account.move.line']._check_company_domain(self.env.companies),
```

Comment:
This will lead to mixing multiple currencies, you need to either
* group by currency and convert to the current main company's currency
* only keep `self.env.company`

Reply:
I'll group by currency and convert to the current main company's currency, as you suggest.